src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticBraceCompletionTests.cs

```diff
@@ -371,6 +371,182 @@ public void Class_OpenBrace_Enter()
             }
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void RecursivePattern()
+        {
+            var code = @"
+class C
+{
+    void M()
+    {
+        _ = this is $$
+    }
+}";
+
+            var expectedBeforeReturn = @"
+class C
+{
+    void M()
+    {
+        _ = this is { }
+    }
+}";
+
+            var expectedAfterReturn = @"
+class C
+{
+    void M()
+    {
+        _ = this is
+        {
+
+        }
+    }
+}";
```

Comment:
not sure i understand this.  where is the 'return' being invoked?

Reply:
This is the same behavior as typing `if (true) {` then `return` today.
When you type the open brace, you get `{ }`. Then if you type `return`, you get:
```
{

}
```